systemtests/src/test/java/io/enmasse/systemtest/isolated/CommonTest.java

```diff
@@ -709,38 +707,19 @@ private void doTestBrokerRemainsOperableOnFullAddress(AddressPlan addressPlan, A
 
         AmqpClient client = getAmqpClientFactory().createQueueClient(addressSpace);
         client.getConnectOptions().setCredentials(user);
-        boolean full = false;
         byte[] bytes = new byte[1024 * 100];
         Random random = new Random();
         int messagesSent = 0;
-        TimeoutBudget timeout = new TimeoutBudget(30, TimeUnit.SECONDS);
         log.info("Start time: get queue full");
-        do {
-            Message message = Message.Factory.create();
-            random.nextBytes(bytes);
-            message.setBody(new AmqpValue(new Data(new Binary(bytes))));
-            message.setAddress(address1.getSpec().getAddress());
-            message.setDurable(true);
-            try {
-                var deliveries = client.sendMessage(address1.getSpec().getAddress(), message).get(5, TimeUnit.SECONDS);
-
-                assertEquals(deliveries.size(), 1);
-                var state = deliveries.get(0).getRemoteState();
-                if (state.getType() == DeliveryStateType.Modified || state.getType() == DeliveryStateType.Rejected) {
-                    full = true;
-                    log.info("broker is full with state {} after sending {} messages", state, messagesSent);
-                }
-                messagesSent++;
-
-                if (timeout.timeoutExpired()) {
-                    log.info("Delivery state is {}", state.getType());
-                    Assertions.fail("Timeout waiting for broker to become full, probably error in test env configuration");
-                }
-            } catch (Exception e) {
-                full = true;
-                log.info("broker is full after sending {} messages, exception", messagesSent, e);
-            }
-        } while(!full);
+        Message message = Message.Factory.create();
+        random.nextBytes(bytes);
+        message.setBody(new AmqpValue(new Data(new Binary(bytes))));
+        message.setAddress(address1.getSpec().getAddress());
+        message.setDurable(true);
+
+        Stream<Message> messageStream = Stream.generate(() -> message);
+        messagesSent = client.sendMessagesCheckDelivery(address1.getSpec().getAddress(), messageStream::iterator, protonDelivery -> protonDelivery.remotelySettled() && !protonDelivery.getRemoteState().getType().equals(DeliveryStateType.Accepted))
```

Comment:
why did you chose !Accepted rather than = Rejected?

Reply:
I was thinking to use rejected but wasn't 100% sure if that would always be the case, so !accepted felt safe. But I agree that rejected would allow matching the expected condition, so that should probably be what should be used.